com.unity.netcode.gameobjects/Runtime/Core/NetworkManager.cs

```diff
@@ -906,6 +918,107 @@ public bool StartHost()
+            return StartServer() ? codeAndAllocation : (null, null);
+        }
+
+        /// <summary>
+        /// Easy relay integration (client): it will initialize the unity services, sign in anonymously, join the relay with the given join code and start the client.
+        /// Note that this will force the use of Unity Transport.
+        /// </summary>
+        /// <param name="joinCode">The join code of the allocation</param>
+        /// <exception cref="ServicesInitializationException"> Exception when there's an error during services initialization </exception>
+        /// <exception cref="UnityProjectNotLinkedException"> Exception when the project is not linked to a cloud project id </exception>
+        /// <exception cref="CircularDependencyException"> Exception when two registered <see cref="IInitializablePackage"/> depend on the other </exception>
+        /// <exception cref="AuthenticationException"> The task fails with the exception when the task cannot complete successfully due to Authentication specific errors. </exception>
+        /// <exception cref="ArgumentException">Thrown if the joinCode has the wrong format.</exception>
+        /// <returns>True if starting the client was successful</returns>
+        public async Task<JoinAllocation> StartClientWithRelay(string joinCode)
+        {
+            await UnityServices.InitializeAsync();
+            if (!AuthenticationService.Instance.IsSignedIn)
+            {
+                await AuthenticationService.Instance.SignInAnonymouslyAsync();
+            }
+            var joinAllocation = await RelayService.Instance.JoinAllocationAsync(joinCode: joinCode);
+            GetUnityTransport().SetRelayServerData(new RelayServerData(joinAllocation, k_ConnectionType));
+            return StartClient() ? joinAllocation : null;
+        }
+
+        private async Task<(string, Allocation)> InitializeAndCreateAllocAsync(int maxConnections)
+        {
+            await UnityServices.InitializeAsync();
+            if (!AuthenticationService.Instance.IsSignedIn)
+            {
+                await AuthenticationService.Instance.SignInAnonymouslyAsync();
+            }
+            Allocation allocation = await RelayService.Instance.CreateAllocationAsync(maxConnections);
+            GetUnityTransport().SetRelayServerData(new RelayServerData(allocation, k_ConnectionType));
+            var joinCode = await RelayService.Instance.GetJoinCodeAsync(allocation.AllocationId);
+            return (joinCode, allocation);
+        }
+
+        private UnityTransport GetUnityTransport()
+        {
+            if (!TryGetComponent<UnityTransport>(out var transport))
+            {
+                transport = gameObject.AddComponent<UnityTransport>();
+            }
+#if UTP_TRANSPORT_2_0_ABOVE
+// TODO: FIXME - Whether UNITY_WEBGL is defined or not, users might want to have a build that supports both modes.
```

Comment:
Same things here (if the answer to both is no then let me know and I will remove the TODO comment).
_I condensed the logic to just set use websockets to true but then this area was what led me to wonder if a user could use websockets on a windows build (or other OS build)..._